chainstate/types/src/gen_block_index.rs

```diff
@@ -81,16 +81,43 @@ impl GenBlockIndex {
         }
     }
 
+    pub fn is_persistent(&self) -> bool {
+        match self {
+            GenBlockIndex::Block(b) => b.is_persistent(),
+            GenBlockIndex::Genesis(..) => true,
+        }
+    }
+
     pub fn chain_transaction_count(&self) -> u128 {
         match self {
             GenBlockIndex::Block(b) => b.chain_transaction_count(),
             GenBlockIndex::Genesis(_) => 0,
         }
     }
+
+    pub fn genesis(chain_config: &ChainConfig) -> Self {
+        Self::Genesis(Arc::clone(chain_config.genesis_block()))
+    }
 }
 
 impl From<BlockIndex> for GenBlockIndex {
     fn from(bi: BlockIndex) -> Self {
         GenBlockIndex::Block(bi)
     }
 }
+
+impl PartialEq for GenBlockIndex {
```

Comment:
As discussed about equality, it should go. Please use this to prevent this from happening again: https://docs.rs/static_assertions/latest/static_assertions/macro.assert_not_impl_any.html